content/python/concepts/command-line-arguments/command-line-arguments.md

```diff
@@ -0,0 +1,101 @@
+
+### Example
+
+This prints the script name followed by a list of the command line arguments passed on to the program.
+
+```python
+import sys
+
+print(f"The program name used to launch me is {sys.argv[0]}.")
+print("I was passed the following arguments:")
+for arg in sys.argv[1:]:
+  print(arg)
+```
+
+If this is named **test.py** it can be launched with the following result:
+
+```bash
+$ test.py --arg1 --arg2 "arg 3"
+The program name used to launch me is test.py.
+I was passed the following arguments:
+--arg1
+--arg2
+arg 3
+```
+
+## Parsing Command Line Arguments With `getopt()`
+
+Using `getopt()` requires importing both the `sys` and `getopt` modules to work. With `getopt`, parameter validation is added to command line parsing. This is done by passing a list of the command line arguments themselves, a string of short (one character) options, and a list of long (full word) options. To indicate that the option requires a value to be passed along with it, the short option is followed by a colon (`:`), and the long option is followed by an equals sign (`=`). So to set up options for "help","argument" and a "value" that requires an additionally passed value, the short options would be a string like this, `"hav:"` and the long options would be a list like this, `["help","argument","value="]`.
+
+### Syntax
+
+```python
+options, values = getopt.getopt(arguments, short_options, long_options)
+```
+
+Where the results of `getopt()` are `options` which is a list of option/value pairs, and `values` which is a list of arguments left after the option list was stripped. The parameters passed to `getopt()` are `arguments`, a list of the arguments as provided by `sys.argv` without the initial program name, the string `short_options` and the list `long_options` as described above.
+
+If `arguments` contains an option that is not in `short_options` or `long_options` then an `getopt.error` exception is thrown.
+
+### Example
+
+This prints the option/value pairs passed as command line arguments.
+
+```python
+import sys, getopt
+
+arguments = sys.argv[1:]
+short_options = "hav:"
+long_options = ["help","argument","value="]
+
+options, values = getopt.getopt(arguments, short_options, long_options)
+
+for o, v in options:
+  print(f"Option is {o}. Value is {v}.")
+```
+
+If this is named **test.py** it can be launched with the following results:
+
+```bash
+$ test.py -a --value=test
+Option is -a. Value is .
+Option is --value. Value is test.
+```
+
+Note that since `-a` wasn't defined as requiring a value passed to it, the corresponding value for the option is empty.
+
+## Other Methods of Parsing the Command Line
+
+There are other methods of parsing command line arguments in Python, which add varying functionality to the task. Some examples include:
+
+- The `argparse` module that's been available since Python 3.2, validates fixed and optional arguments, and offers a default help message displaying the accepted arguments.
```

Comment:
```suggestion
- The `argparse` module that's been available since Python 3.2, validates fixed and optional arguments and offers a default help message displaying the accepted arguments.
```